test/qpy_compat/test_qpy.py

```diff
@@ -580,6 +576,22 @@ def assert_equal(reference, qpy, count, bind=None):
         )
         sys.stderr.write(msg)
         sys.exit(1)
+    # Check deprecated bit properties, if set.  The QPY dumping code before Terra 0.23.2 didn't
+    # include enough information for us to fully reconstruct this, so we only test if newer.
+    if version_parts >= (0, 23, 2) and isinstance(reference, QuantumCircuit):
```

Comment:
I think it's necessary because the _dumping_ code was wrong before 0.23.2. If the version under test (i.e. the one that generated the files) is pre-0.23.2, even this branch will fail to pass these assertions because the information needed to do the roundtrip completely just isn't present in the QPY file.

Reply:
Ah good call, that's fair I had forgotten about the serialization side changes. Then yeah this is definitely needed.